src/Cake.Core/ICakeTaskInfo.cs

```diff
@@ -27,6 +27,6 @@ public interface ICakeTaskInfo
         /// Gets the task's dependencies.
         /// </summary>
         /// <value>The task's dependencies.</value>
-        IReadOnlyList<string> Dependencies { get; }
+        IReadOnlyList<CakeTaskDependency> Dependencies { get; }
```

Comment:
This is a breaking change, for any script / module / addin that uses/implements this interface.

Benefit needs to be evaluated by @cake-build/team